app/(main)/league/all/page.test.tsx

```diff
@@ -1,124 +1,235 @@
 
-  xtest('should render "You are not enrolled in any leagues" message when no leagues are found', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+  test('should render "You are not enrolled in any leagues" message when no leagues are found', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
+      user: { id: '123', leagues: [] },
+      allLeagues: [],
+    });
     mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+
     render(<Leagues />);
 
-    await waitFor(() => {
-      expect(
-        screen.getByText('You are not enrolled in any leagues'),
-      ).toBeInTheDocument();
+    await waitForElementToBeRemoved(() => screen.getByTestId('global-spinner'));
+
+    expect(screen.getByTestId('no-leagues-message')).toBeInTheDocument();
+  });
+
+  test('should display GlobalSpinner while loading data', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValueOnce({
+      user: { id: '123', leagues: [] },
+      allLeagues: [],
     });
+
+    render(<Leagues />);
+
+    expect(screen.getByTestId('global-spinner')).toBeInTheDocument();
   });
 
-  xtest('should display GlobalSpinner while loading data', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+  test('should not display GlobalSpinner after loading data', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
+      user: { id: '123', leagues: [] },
+      allLeagues: [
+        {
+          leagueId: '123',
+          leagueName: 'Test League',
+          logo: 'logo.png',
+          participants: ['123456', '78'],
+          survivors: ['123456', '78', '9'],
+        },
+      ],
+    });
     mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+
     render(<Leagues />);
 
-    await waitFor(() => {
-      expect(screen.getByTestId('global-spinner')).toBeInTheDocument();
-    });
+    await waitForElementToBeRemoved(() => screen.getByTestId('global-spinner'));
+
+    expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
   });
-  xtest('should not display GlobalSpinner after loading data', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+
+  test('should handle form submission to join a league', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
+      user: { id: '123', leagues: [] },
+      allLeagues: [
+        {
+          leagueId: '123',
+          leagueName: 'Test League',
+          participants: [],
+          survivors: [],
+        },
+      ],
+    });
+
+    mockGetAllLeagues.mockResolvedValueOnce([
+      {
+        leagueId: '123',
+        leagueName: 'Test League',
+        participants: [],
+        survivors: [],
+      },
```

Comment:
logo is missing from here.  can even be an empty string.  just want to make sure we are showing the correct data structure.

Reply:
Added in `logo`.